server/src/main/kotlin/fi/oph/kitu/kotoutumiskoulutus/KoealustaMappingService.kt

```diff
@@ -10,24 +12,45 @@ import java.time.Instant
 class KoealustaMappingService(
     private val oppijanumeroService: OppijanumeroService,
 ) {
-    fun suorituksetToEntity(suorituksetResponse: KoealustaSuorituksetResponse): List<KielitestiSuoritus> =
-        suorituksetResponse.users.flatMap { user ->
-            val oppijanumero = getOppijanumero(user)
-            user.completions.map { completion ->
-                completionToEntity(
-                    user,
-                    oppijanumero,
-                    completion,
-                )
+    fun convertToEntity(suorituksetResponse: KoealustaSuorituksetResponse): List<KielitestiSuoritus> {
+        val exceptions = mutableListOf<OppijanumeroException>()
+
+        val entity =
+            suorituksetResponse.users.flatMap { user ->
+                val oppijanumero =
+                    try {
+                        oppijanumeroService.getOppijanumero(toOppija(user))
+                    } catch (ex: OppijanumeroException) {
+                        exceptions.add(ex)
+                        null
+                    }
+
+                user.completions.map { completion ->
+                    completionToEntity(
+                        user,
+                        oppijanumero,
+                        completion,
+                    )
+                }
             }
+
+        return if (exceptions.isNotEmpty()) {
+            throw KoealustaMappingServiceException(
+                "Unable to convert into list of KielitestiSuoritus, because there were ${exceptions.size} exceptions from oppijanumero-service.",
+                exceptions,
+            )
+        } else {
+            entity
         }
```

Comment:
Hieman nillitykseen kallistuva ehdotus:

```suggestion
        if (exceptions.isNotEmpty()) {
            throw KoealustaMappingServiceException(
                "Unable to convert into list of KielitestiSuoritus, because there were ${exceptions.size} exceptions from oppijanumero-service.",
                exceptions,
            )
        }
        return entity
```